### ISSUE TITLE
Enhancement: Underground biomes [1.18.x]

### ISSUE THREAD

**Issue by Royal-Jelly** [alcatrazEscapee/hexlands#5]
I have some questions related underground biomes in future 1.18+ ports

Should the underground biomes to span only to the border of the same biome hex?
or would it be more feasible for the underground biome placement to be unrestricted?

also if both is are doable may it be a config option?

Thank you for your time ~

**Comment by alcatrazEscapee** [comment]
This is an excellent question, and it's largely going to depend on how it ends up being implemented. As you know, 1.18 rewrote much of the underlying world generation that Hex Lands was using, namely biome generation is... functionally entirely different, and thus going to be quite difficult to figure out how to reconcile that with new vanilla, since the notion of "biomes" has changed.

**Comment by alcatrazEscapee** [comment]
As of [the 1.18.2 release](https://github.com/alcatrazEscapee/hexlands/releases/tag/v3.0.0), and further investigating vanilla's biome and chunk generation:

- Splitting biome generation such that underground biomes are generated "per usual" and above ground biomes are generated via hexagons is not feasible, and would require rewriting large parts of vanilla in order to accomplish this. There is no notion of what an actual underground biome is in the way that hexlands coerces biome generation into hexagons.
- As a result of this, aquifers, and other changes (notably being that height generation is almost completely transparent, and estimating the height of any given "biome" isn't feasible), hex borders have been changed as if `border_extends_to_bedrock` from 1.16 is always true, effectively segmenting the underground biomes as well.

In short, hexagons have been made to work for the overworld biomes. Whatever has happened underground is just a consequence of the former, and untangling the two is devilishly complicated.
